### ISSUE TITLE
How to do Full Text indexing and search on the below json Document ?

### ISSUE THREAD

**Issue by ansarihaseb** [arangodb/arangodb#1658]
{
	"batters":
		{
			"batter":
				[
					{ "id": "1001", "type": "Regular" },
					{ "id": "1002", "type": "Chocolate" },
					{ "id": "1003", "type": "Blueberry" },
					{ "id": "1004", "type": "Devil's Food" }
				]
		},
	 "topping":
		[
			{ "id": "5001", "type": "None" },
			{ "id": "5002", "type": "Glazed" },
			{ "id": "5005", "type": "Sugar" },
			{ "id": "5007", "type": "Powdered Sugar" },
			{ "id": "5006", "type": "Chocolate with Sprinkles" },
			{ "id": "5003", "type": "Chocolate" },
			{ "id": "5004", "type": "Maple" }
		]
    }

Basically to have full-text search here I would need to do indexing on "batters.batter" and also on "batters.topping" i.e. on two attributes. How to handle this kind of full text searching. Please explain on the method and I would to implement my search through REST API.  Thanking You in advance.

**Comment by CoDEmanX** [comment]
AFAIK it's not possible to have both in a single fulltext index, because only the immediate child objects are checked. It would require two levels in your example. But this is surely something that should be added ("composite indexes"?)

**Comment by dothebart** [comment]
as https://docs.arangodb.com/devel/HttpIndexes/Fulltext.html explains, and as you state you need two fulltext indices, one on batters.batter, and one on batters.topping. @CoDEmanX is sort of correct, `[{foo:"bar"}]` will work, but one more layer `[{foo: {morefoo: "bar"}}]` won't.

However, there currently is no way to combine two fulltext indices in Arangodb. You have to do basically two separate queries to it.
You could however [use two combined](https://docs.arangodb.com/devel/AqlExamples/Join.html#horizontal_lists) [AQL queries](https://docs.arangodb.com/devel/IndexHandling/Fulltext.html#accessing_fulltext_indexes_from_the_shell) to get the result in one chunk.

**Comment by ansarihaseb** [comment]
Combine two AQL queries ? What does it literally means and How to do it with API........still confused

**Comment by ansarihaseb** [comment]
@dothebart can you give one example how to get the search result in one chunk ? It's urgent I have to implement the full-text search in my application. Thanking You

**Comment by dothebart** [comment]
closing and continuing on [stackoverflow](http://stackoverflow.com/questions/34792722/how-to-do-full-text-indexing-and-search-on-the-below-json-document-in-arangodb)
